### PR TITLE
pkg/ui: performance fixes on redesigned tracez page

### PR DESCRIPTION
**Pull request by benbardin**
Followup to #88181 

Part of https://github.com/cockroachdb/cockroach/issues/83679

Release note: None

### CODE REVIEW

File: pkg/ui/workspaces/cluster-ui/src/tracez/snapshot/snapshotPage.tsx [hunk 1]
```diff
@@ -62,27 +63,26 @@ export const SnapshotPage: React.FC<SnapshotPageProps> = props => {
...
+  const columnTitle = match.params.columnTitle || "";
 
   const snapshotIDStr = getMatchParamByName(match, "snapshotID");
 
   useEffect(() => {
     refreshSnapshots();
-  });
+  }, [refreshSnapshots]);
+
+  const snapArray = snapshots?.snapshots;
```
Comment (sjbarag) [inline pkg/ui/workspaces/cluster-ui/src/tracez/snapshot/snapshotPage.tsx:76]: Oooh good idea!

**Review by sjbarag — APPROVED** [review]
Niiice. Thanks again for tackling this so quickly @benbardin 🙂

**Comment by benbardin** [comment]
test failure is unrelated

bors r+

**Comment by benbardin** [comment]
> Niiice. Thanks again for tackling this so quickly @benbardin 🙂

You're very welcome! Happy to do it when you literally wrote out the answers for me 😂

**Pull request merged**